util/chunk/iterator.go

```diff
@@ -105,13 +105,15 @@ func NewIterator4Chunk(chk *Chunk) *Iterator4Chunk {
 
 // Iterator4Chunk is used to iterate rows inside a chunk.
 type Iterator4Chunk struct {
-	chk    *Chunk
-	cursor int
+	chk     *Chunk
+	cursor  int
+	numRows int
 }
 
 // Begin implements the Iterator interface.
 func (it *Iterator4Chunk) Begin() Row {
-	if it.chk.NumRows() == 0 {
+	it.numRows = it.chk.NumRows()
+	if it.numRows == 0 {
```

Comment:
Is there any use case that we are iterating the chunk while appending into it? I guess this is not allowed, just to make sure.

Reply:
@zz-jason